Initial governance document for the Binder project (#100)
jupyterhub/team-compass: docs/governance.rst

The following sections describe governance and high-level principles
surrounding the Binder Project.

Project mission
~~~~~~~~~~~~~~~

The mission of the Binder Project is to create, advance, and promote
open technology that makes it easy for people to connect their data
science communication, education, and scientific reproducibility work
with computational environments which can run their work.

The Binder Project pursues this mission by advancing the tools
surrounding BinderHub, an open-source tool for hosting ad-hoc
environments in the cloud. The project also serves as the primary
maintainer of ``mybinder.org``, a public service and demonstration of
the BinderHub technology.

Membership
~~~~~~~~~~

Membership in the Binder Project is open to anyone who is nominated by
an existing member. Members form the trusted group of individuals who
manage the Binder Project.

Membership comes with `privileges`_ and `responsibilities`_.

Currently, the Binder Project team overlaps 100% with `the BinderHub
team`_. However, these two groups may diverge in their membership over
time.

The authoritative list of members and their roles (alphabetical sorting):
* Tim Head (red)

> * Min Ragan-Kelley (team leader)

Reviewer: ```suggestion
* Min Ragan-Kelley (team leader)
* Erik Sundell (blue)
```

Reviewer: I don't understand how to avoid cluttering a line above or below though using this "review comment suggestions" feature.

Author: I think this worked?! I clicked commit and it seems to have done what I was expecting it to do.

> * Your Name (red or blue)

Reviewer: ```suggestion
* Erik Sundell (blue)
* Your Name (red or blue)
```

The following sections describe governance and high-level principles
surrounding the Binder Project.

Project mission
~~~~~~~~~~~~~~~

The mission of the Binder Project is to create, advance, and promote
open technology that makes it easy for people to connect their data
science communication, education, and scientific reproducibility work
with computational environments which can run their work.

The Binder Project pursues this mission by advancing the tools
surrounding BinderHub, an open-source tool for hosting ad-hoc
environments in the cloud. The project also serves as the primary
maintainer of ``mybinder.org``, a public service and demonstration of
the BinderHub technology.

Membership
~~~~~~~~~~

Membership in the Binder Project is open to anyone who is nominated by
an existing member. Members form the trusted group of individuals who
manage the Binder Project.

Membership comes with `privileges`_ and `responsibilities`_.

Currently, the Binder Project team overlaps 100% with `the BinderHub
team`_. However, these two groups may diverge in their membership over
time.

The authoritative list of members and their roles (alphabetical sorting):
* Tim Head (red)

> * Min Ragan-Kelley (team leader)

Reviewer 2: ```suggestion
* Yuvi Panda (red)
```

Author: I think we need more guidance from @consideRatio because your suggestion would replace Min with your name. Somehow Erik managed to add himself instead of replacing.

Reviewer: The behavior of the suggestion code block seem to be that it *replaces the row with whatever is inside the suggestion block*.

It also seem to fail to realize that the original line is still there.

Binder Project Governance
=========================

The following sections describe governance and high-level principles
surrounding the Binder Project.

The Binder Project is a member of Project Jupyter, which is a fiscally
sponsored project of NumFocus, a US 501c3 non-profit.

Project mission
~~~~~~~~~~~~~~~

The mission of the Binder Project is to create, advance, and promote
open technology that makes it easy for people to connect their data
science communication, education, and scientific reproducibility work
with computational environments which can run their work.

The Binder Project pursues this mission by advancing the tools
surrounding BinderHub, an open-source tool for hosting ad-hoc
environments in the cloud. The project also serves as the primary
maintainer of ``mybinder.org``, a public service and demonstration of
the BinderHub technology.

Membership
~~~~~~~~~~

> Membership in the Binder Project is open to anyone who is nominated by

Reviewer 3: There should be some mention of inclusion. Nomination by existing members tends to reinforce the status quo unless conscious outreach is done. Outreachy is a good start but more should be done.

Author: 👍  do you have a suggestion for a sentence we can use here?

For me this document is more like a legal reference document so right now it is very brief on philosophy/intentions :-/

My best idea right now is to take https://github.com/jupyterhub/team-compass/issues/42#issuecomment-451458207 and turn it into an issue to get the ball rolling. Then insert a link to that issue here and then replace it with a link to the outcome of the issue once we have it. Suggested sentence: "To find out more about the process and how to get started see link-to-issue."

Reviewer 4: While it could get a mention in the heading here, I think it definitely warrants an explicit point in the "responsibilities" below.

Reviewer 5: I like the idea of explicitly making inclusion and increasing the diversity of the community part of the responsibilities of being a "red" team member (or just a member in general)

an existing member. Members form the trusted group of individuals who
manage the Binder Project.

Membership comes with `privileges`_ and `responsibilities`_.

> Currently, the Binder Project team overlaps 100% with `the BinderHub

Reviewer 3: Shouldn't all members of the current BinderHub team be added to the list of names?

Reviewer 5: @betatim I slightly edited your top-level comment to **bold** the sentence where you ask people to add themselves via suggestions, I hope that's ok. I missed it too :-P

> Currently, the Binder Project team overlaps 100% with `the BinderHub

Reviewer 3: If so, any reason that we are not adding the existing team?

> Binder Project Governance

Reviewer 5: Maybe the file should be called `governance_binder.rst`? I don't think that jupyterhub has its own governance structure, but this might help disambiguate things a bit

Project mission
~~~~~~~~~~~~~~~

The mission of the Binder Project is to create, advance, and promote
open technology that makes it easy for people to connect their data
science communication, education, and scientific reproducibility work
with computational environments which can run their work.

The Binder Project pursues this mission by advancing the tools
surrounding BinderHub, an open-source tool for hosting ad-hoc
environments in the cloud. The project also serves as the primary
maintainer of ``mybinder.org``, a public service and demonstration of
the BinderHub technology.

Membership
~~~~~~~~~~

Membership in the Binder Project is open to anyone who is nominated by
an existing member. Members form the trusted group of individuals who
manage the Binder Project.

Membership comes with `privileges`_ and `responsibilities`_.

Currently, the Binder Project team overlaps 100% with `the BinderHub
team`_. However, these two groups may diverge in their membership over
time.

The authoritative list of members and their roles (alphabetical sorting):
* Tim Head (red)
* Min Ragan-Kelley (team leader)
* Yuvi Panda (red)
* Erik Sundell (blue)
* Carol Willing (red)

> * Your Name (red or blue)

Reviewer 5: ```suggestion
* Chris Holdgraf (red)
* Your Name (red or blue)
```

I'd also be +1 on making this alphabetized

Author: Adding you by hand so that you appear in the right place. The list should be by ABC but currently there is a mistake (fixed).

> Team privileges

Reviewer 5: These apply to both red+blue team members unless otherwise stated, right?

Reviewer 6: More broadly I'm just not sure where these roles came from?  I don't remember discussing this at the last meeting?

Author: These came from discussions in #42. There was no in person discussions.

~~~~~~~~~~~~~~~

-  recognition
-  team members can speak for the project in public
-  merge privileges on the project repositories
-  red members can nominate other individuals to team membership

Team responsibilities
~~~~~~~~~~~~~~~~~~~~~

Each member of the Binder Project team is expected to spend a
"reasonable" (intentionally vague) amount of time contributing to the
technology used by Project Binder as well as maintaining
``mybinder.org``. This might mean contributing features, making patches,
responding to issues, reviewing PRs, teaching about BinderHub, making
organizational connections, welcoming new contributors, or evangelizing
and helping others deploy BinderHub. They should be receptive to
comments and ideas on the GitHub issues or in the Binder gitter channel.
They should strive to make project decisions that reflect these
community opinions.

> 

Reviewer 5: ```suggestion

In addition, red team members are charged with growing
the Binder community and ensuring that we create an open and inclusive
culture for users and future team members.
```

Author: I didn't take over this suggestion as is. Instead I added 09325b4bd426e9687895bdf0d31b441978c1da16. What do you think of that?

This list is non-complete.

Team roles
~~~~~~~~~~

-  Team leader. This is a person who has impasse-breaking authority to
   make decisions if the team can not agree. This authority should
   be used exceedingly sparingly. The team leader is expected to grow
   team members so that they can pass on their position to them.
-  Red member. You spend a large part of your time coordinating and
   promoting the Binder Project as well as the service at
   ``mybinder.org``. You are expected to coordinate your actions with
   other red members so that the project presents a coherent position
   towards the outside. You are expected to prepare and attend the
   monthly team calls. Being a red member is not a life sentence. We
   expect red members to teach and grow other members so that they can
   pass their organizational responsibilities on to them. This encourages

>    is to increase diversity, reduce burn-out and allow team members to

Reviewer 4: ```suggestion
   us to increase diversity, reduce burn-out and allow team members to
```

   shift their focus in reaction to changing life circumstances.
-  Blue Member. This is where the journey of a new team member starts.
   You spend a large
   part of your time facilitating contributions from others to the
   project (e.g. reviewing work, answering questions, maintaining
   project infrastructure). We expect blue members to teach and grow
   contributors that can become blue members. This encourages diversity
   of the team. After building a good understanding of the technology
   and social structure of the project you can change to a red member by
   volunteering for tasks which a red member is expected to perform.
   After a while of doing so the team will ask if you want to become a
   red member and, if you agree, recognise you as such at a monthly team
   meeting.

> There are no other specific roles, but we may revisit this in the

Reviewer 5: What about one more status such as "inactive team member" that would be a place for people who want to take a break, move on to other things etc, and would signal that they're still "members" of the community, just not actively working on it now (and make a note that anyone is welcome to become active again if they are willing to take on the responsibilities once more)

Reviewer 7: I really like the JOSS strategy for naming editors who have moved on: ["editors emeritus"](https://joss.theoj.org/about#editorial_board). It implies that they are members of the community, people you can potentially reach out to for general questions or mentorship (though not required), but also indicates that they are no longer involved in the day-to-day work of the project

Author: Do you feel it makes people nervous to become "emeritus" because it is seen as a one way street?

Reviewer 7: That is a fair point @betatim; "emeritus" does imply a one-way street, and might not be appropriate to use instead of the "inactive" status. 

Would there be any reason to consider having both? If someone has moved-on and doesn't intend to come back any time in the near future but is willing to mentor it could be fitting? (However, adding another designation might just complicate matters?)

Author: Bunch of words I could think of that are related, maybe one works? retired, inactive, emeritus, paused, graduates, alumni, moved on, green (just another colour)

What do you think of the following role description:
"""
* xxx member, your life situation has changed so that you prefer to (temporarily) not take on the rights and responsibilities of being an active member. You can return to active membership by resuming your community activities. Your return will be announced at the next monthly team meeting.
"""

Reviewer 7: That looks good! And I like the idea of sticking with colours - there are fewer connotations / previous associations than any of the other terms

Author: Done.

Red + blue + green = white! We need all the roles to create a bright light :)💡

future.

How decisions are made
~~~~~~~~~~~~~~~~~~~~~~

The Binder Project team will make decisions `as colleagues`_ and by

> attempting to reach consensus among team members. If this is not

Reviewer 5: ```suggestion
attempting to reach consensus among team members (similar to a [lazy consensus](http://en.osswiki.info/concepts/lazy_consensus) model that
encourages active participation from team members). If this is not
```

Reviewer 5: oops, yep, done

Author: Done by hand afterall.

possible, then the team leader can use their power to make a decision.

There is currently no formal specification for this decision process.

Team size
~~~~~~~~~

There is no limit to the size of the team.

There is no hard cap on the number of senior members that can exist,
though this will be re-evaluate if the group becomes big enough to be

> unwieldy. We see the ideal number of senior members to be around 6-10

Reviewer 5: ```suggestion
unwieldy. We see the ideal number of red team members to be around 6-10
```
